### ISSUE TITLE
Feature request: Send a break to serial port on connection

### ISSUE THREAD

**Issue by pfalcon** [wtarreau/bootterm#6]
I looked around, and was amazed by the fact that there doesn't seem to be a serial terminal which can send a BREAK to serial port on connection (only using escape command triggered by a user). Sending a BREAK on connection is the default behavior of the ser2net tool, and sought a way to emulate it. And I couldn't! As bootterm is a new tool, aspiring to improve on its predecessors, maybe it'll be interested to add it...

To elaborate why this may be useful - various devices may treat UART BREAK as analog of a "reset" signal. This is e.g. the case for CMSIS bootloader boards (many ARM boards, a random example is NXP FRDM-K64F board). So, if such a board is programmed with a simple "start, print something, stop" embedded application (common situation during initial stages of development and/or experimenting), the by sending BREAK on connection, you will cause the programmed app to re-run and get output. Whereas otherwise, you wouldn't see any output on connection, and would need to reach for a reset button, which is quite boring.

**Comment by wtarreau** [comment]
I was about to say that I was thinking it could be seriously annoying a feature to have by default, as I'm having a board which systematically resets when I connect to it, resulting in code losses when there are connectivity issues or if I quit/reconnect. But your case sounds like just a programmable reset to me and that makes sense. I'm even thinking that it could be nice to support sending the break and quitting to put it in scripts. I'll see what I can do.

**Comment by pfalcon** [comment]
Thanks for response!

I definitely don't ask it to be enabled *by default* (I just mentioned that ser2net has it like that by default, which is quite a cowboy approach, I would agree ;-) ). Just a command-line switch to send it on connect, e.g. `--send-break`.

> I'm even thinking that it could be nice to support sending the break and quitting to put it in scripts. I'll see what I can do.

Might be. I actually tried to follow that approach, by sending break using a Python script, and then running picocom (all in a shell script), but that was too slow, a big chunk of initial output didn't get to picocom. (It's only after that I remembered that I read about bootterm and proceeded to check if it has builtin option for that. Seeing that nope, I just posted the issue, but didn't actually try bootterm.)
